src/applications/edu-benefits/components/SchoolSelectField.jsx

```diff
@@ -0,0 +1,216 @@
+      institutionQuery: this.props.institutionQuery,
+      page
+    });
+  }
+  render() {
+    const {
+      currentPageNumber,
+      facilityCodeSelected,
+      formContext,
+      institutionQuery,
+      institutions,
+      institutionSelected,
+      pagesCount,
+      searchInputValue,
+      searchResultsCount,
+      showInstitutions,
+      showInstitutionsLoading,
+      showPagination,
+      showPaginationLoading,
+    } = this.props;
+
+    const fieldsetClass = classNames('search-select-school-fieldset');
+    const legendClass = classNames('legend-label');
+
+    if (formContext.reviewMode) {
+      const {
+        city,
+        name,
+        state
+      } = institutionSelected;
+
+      return (
+        <div>
+          <p>{name}</p>
+          <p>{`${city}, ${state}`}</p>
+        </div>
+      );
+    }
+
+    return (
+      <fieldset className={fieldsetClass}>
+        <legend className={legendClass}>
+          {'Search school names'}
+        </legend>
+        <div>
+          <div className="search-controls">
+            <Element name="schoolSearch"/>
+            <input
+              onChange={this.handleSearchInputChange}
+              onKeyDown={this.handleSearchInputKeyDown}
+              type="text"
+              value={searchInputValue}/>
+            <button
+              className="search-schools-button usa-button-primary"
+              onClick={this.handleSearchClick}>
+              {'Search Schools'}
+            </button>
+          </div>
+          {showInstitutions && <div>
+            {`${searchResultsCount} results for ${institutionQuery}`}
+            {institutions.map(({ city, facilityCode, name, state }, index) => (
+              <div key={index}>
+                <div className="radio-button">
+                  <input
+                    autoComplete="false"
+                    checked={facilityCodeSelected === facilityCode}
+                    id={`radio-buttons-${index}`}
+                    name={name}
+                    type="radio"
+                    onKeyDown={this.onKeyDown}
+                    onChange={() => this.handleOptionClick({ city, facilityCode, name, state })}
+                    value={facilityCode}/>
+                  <label
+                    id={`institution-${index}-label`}
+                    htmlFor={`radio-buttons-${index}`}>
+                    <span className="institution-name">{name}</span>
+                    <span className="institution-city-state">{`${city}, ${state}`}</span>
+                  </label>
+                </div>
+              </div>))
+            }
+          </div>
+          }
+          {showInstitutionsLoading && <div>
+            <LoadingIndicator message={`Searching ${institutionQuery}...`}/>
+          </div>
+          }
+          {showPaginationLoading && <div>
+            <LoadingIndicator message={`Loading page ${currentPageNumber} results for ${institutionQuery}...`}/>
```

Comment:
Might be gold plating to have two different loading states but wanted to have the option.